apps/files/list.php

```diff
@@ -22,10 +22,14 @@
  *
  */
 
+use OCP\Share\IManager;
+
 $config = \OC::$server->getConfig();
 $userSession = \OC::$server->getUserSession();
 // TODO: move this to the generated config.js
-$publicUploadEnabled = $config->getAppValue('core', 'shareapi_allow_public_upload', 'yes');
+/** @var IManager $shareManager */
+$shareManager = \OC::$server->get(IManager::class);
+$publicUploadEnabled = $shareManager->shareApiLinkAllowPublicUpload();
```

Comment:
I couldn't find any usage of the data attribute where this value is assigned to but from a compatibility perspective we should probably preserve the yes/no values here:

```suggestion
$publicUploadEnabled = $shareManager->shareApiLinkAllowPublicUpload() ? 'yes' : 'no';
```